design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+The user wants to answer the following questions:
+
+* When will my build finish?
+    * Soon (should I keep watching it) or later ("I'll check back")
+* Is there anything I should know about my build right now? (e.g. tests have failed, warnings)
+* What is Gradle doing right now?
+* Is my build successful?
+* What command did I execute?
+
```

Comment:
I believe the first point should be added. I hesitate to consider the latter two. I personally get the sense that users really do not care at all about how Gradle achieve it's result. If they do, that's what build scans are for!